sonoff/xsns_36_mgc3130.ino

```diff
@@ -0,0 +1,651 @@
+/*
+  xsns_91_MGC3130.ino - Support for I2C MGC3130 Electric Field Sensor for Sonoff-Tasmota
+
+  Copyright (C) 2018  Christian Baars & Theo Arends
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are met:
+
+  - Redistributions of source code must retain the above copyright notice,
+    this list of conditions and the following disclaimer.
+  - Redistributions in binary form must reproduce the above copyright notice,
+    this list of conditions and the following disclaimer in the documentation
+    and/or other materials provided with the distribution.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+  POSSIBILITY OF SUCH DAMAGE.
+*/
+
+//#define USE_MGC3130
+
+#ifdef USE_I2C
+#ifdef USE_MGC3130
+
+#define XSNS_36 36
+
+/*********************************************************************************************\
+ * MGC3130 - Electric Field Sensor
+ *
+ * Adaption for TASMOTA: Christian Baars
+ * based on various implementations from Pimoroni, jspark311, hoverlabs and scjurgen
+ *
+ * I2C Address: 0x42
+ *
+ * Wiring: SDA/SCL as usual plus RESET and TRANSFER -> 4 Wires
+\*********************************************************************************************/
+
+#warning **** MGC3130: It is recommended to disable all unneeded I2C-drivers ****
```

Comment:
Just a question: Why is that? It uses the I2C a lot?

Reply:
It is only a hint, that a lot of traffic on the I2C-bus can be generated and if something does not work, this could be a place to start for bug hunting.
I did not run into trouble here, but if I remember correctly, somewhere in the datasheets from MICROCHIP there was a recommendation to reduce the bus traffic as good as as possible (at least I interpreted it that way). Maybe it is a bit exaggerated ;)